bottlerocket/samples/Makefile.toml

```diff
@@ -0,0 +1,248 @@
+    if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+        AMI_VERSION=$(echo ${STARTING_VERSION} | tr -d "v")
+    else
+        AMI_VERSION="latest"
+    fi
+    aws ssm get-parameter \
+        --region ${AWS_REGION} \
+        --name "/aws/service/bottlerocket/${VARIANT}/$(echo ${ARCHITECTURE} | sed -e 's/aarch64/arm64/g')/${AMI_VERSION}/image_id" \
+        --query Parameter.Value --output text 
+fi
+''']}
+
+[tasks.gpu.env]
+PSEUDO_GPU = {script=['''
+if [ $(echo ${VARIANT} | tr "-" " " | awk '{print $NF}') = "nvidia" ]; then
+    echo true
+else
+    echo false
+fi
+''']}
+
+[tasks.mgmt-cluster-kubeconfig-base64.env]
+MGMT_CLUSTER_KUBECONFIG_BASE64 = {script=['''
+if [ -n "${MGMT_CLUSTER_KUBECONFIG_PATH}" ]; then
+    cat ${MGMT_CLUSTER_KUBECONFIG_PATH} | base64
+fi
+''']}
+
+[tasks.set-pseudo-env]
+dependencies = ["set-agent-images", "cluster-name", "gpu", "metadata-url", "ova-name", "bottlerocket-ami-id"]
+
+[tasks.set-env]
+dependencies = ["set-pseudo-env", "mgmt-cluster-kubeconfig-base64"]
+
+[tasks.set-env.env]
+CLUSTER_NAME = { value = "${PSEUDO_CLUSTER_NAME}", condition = { env_not_set = ["CLUSTER_NAME"] } }
+GPU = { value = "${PSEUDO_GPU}", condition = { env_not_set = ["GPU"] } }
+METADATA_URL = { value = "${PSEUDO_METADATA_URL}", condition = { env_not_set = ["METADATA_URL"] } }
+OVA_NAME = { value = "${PSEUDO_OVA_NAME}", condition = { env_not_set = ["OVA_NAME"] } }
+BOTTLEROCKET_AMI_ID = { value = "${PSEUDO_BOTTLEROCKET_AMI_ID}", condition = { env_not_set = ["BOTTLEROCKET_AMI_ID"] } }
+
+[tasks.instance-types]
+dependencies = ["set-env"]
+
+[tasks.instance-types.env]
+INSTANCE_TYPES = {script=['''
+if [ -z ${INSTANCE_TYPES} ]; then
+    if [ ${ARCHITECTURE} = "x86_64" ]; then
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g4dn.xlarge\"]"
+        else
+            echo "[\"m5.large\"]"
+        fi
+    else
+        if [ ${GPU} = "true" ]; then
+            echo "[\"g5g.xlarge\"]"
+        else
+            echo "[\"m6g.large\"]"
+        fi
+    fi
+else
+    echo ${INSTANCE_TYPES}
+fi
+''']}
+
+[tasks.create-inner]
+dependencies = ["set-env", "instance-types"]
+script_runner = "bash"
+script = '''
+eval "cat > ${OUTPUT_FILE} << EOF
+$(< ${CLUSTER_TYPE}/${TEST_TYPE}.yaml)
+EOF
+" 2> /dev/null
+
+echo "${OUTPUT_FILE}"
+'''
+
+[tasks.create-inner.env]
+OUTPUT_FILE = {script=['''
+if [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "migration" ]; then
+    echo "output/${CLUSTER_NAME}-migration.yaml"
+elif [ $(echo ${TEST_TYPE} | tr "-" " " | awk '{print $2}') = "workload" ]; then
+    echo "output/${CLUSTER_NAME}-workload.yaml"
+else
+    echo "output/${CLUSTER_NAME}.yaml"
+fi
+''']}
+
+[tasks.create]
+script = '''
+#!@duckscript
+set_env TEST_TYPE ${1}
+cm_run_task ${1}
+cm_run_task create-inner
+'''
+
+[tasks.install-controller]
+script = "cli install --controller-uri ${CONTROLLER_IMAGE_URI}"
+
+[tasks.add-secrets]
```

Comment:
Let's break this into 2 targets. One for vSphere and one for AWS creds. You can use `conditions` to only have the vSphere secret be created for VMware variants.